diff -pruN genders-1.4-1/src/Libgenders/Makefile.am genders-1.4-1-hack/src/Libgenders/Makefile.am
--- genders-1.4-1/src/Libgenders/Makefile.am	2005-04-20 05:17:27.000000000 +0530
+++ genders-1.4-1-hack/src/Libgenders/Makefile.am	2006-06-26 20:21:29.000000000 +0530
@@ -39,7 +39,7 @@ Makefile.xs: Makefile.PL Libgenders.xs L
 	if ! test -a Libgenders.xs; then \
 		cp $(srcdir)/Libgenders.xs .; \
 	fi
-	$(PERL) Makefile.PL $(MAKEMAKERFLAGS)
+	$(PERL) Makefile.PL $(MAKEMAKERFLAGS) $(EXTRA_PERLFLAGS)
 
 test: Makefile.xs 
 	$(MAKE) -f Makefile.xs test
diff -pruN genders-1.4-1/src/Libgenders/Makefile.in genders-1.4-1-hack/src/Libgenders/Makefile.in
--- genders-1.4-1/src/Libgenders/Makefile.in	2005-10-04 01:25:17.000000000 +0530
+++ genders-1.4-1-hack/src/Libgenders/Makefile.in	2006-06-26 20:21:48.000000000 +0530
@@ -364,7 +364,7 @@ Makefile.xs: Makefile.PL Libgenders.xs L
 	if ! test -a Libgenders.xs; then \
 		cp $(srcdir)/Libgenders.xs .; \
 	fi
-	$(PERL) Makefile.PL $(MAKEMAKERFLAGS)
+	$(PERL) Makefile.PL $(MAKEMAKERFLAGS) $(EXTRA_PERLFLAGS)
 
 test: Makefile.xs 
 	$(MAKE) -f Makefile.xs test
