diff -pruN lam-7.1.2/configure lam-7.1.2-hacked/configure
--- lam-7.1.2/configure	2006-03-11 03:01:14.000000000 +0530
+++ lam-7.1.2-hacked/configure	2006-06-26 16:54:13.000000000 +0530
@@ -4396,32 +4396,6 @@ echo "${ECHO_T}$co_result" >&6
 CFLAGS="$co_result"
 
 
-# config/lam_ansi_protos.m4
-
-# Determine if compiler and options accepts ANSI prototypes.
-echo "$as_me:$LINENO: checking whether the C compiler ($CC $CFLAGS) accepts ANSI prototypes" >&5
-echo $ECHO_N "checking whether the C compiler ($CC $CFLAGS) accepts ANSI prototypes... $ECHO_C" >&6
-cat > conftest.c <<EOF
-static double foo(double bar) { return(2.0*bar); }
-int main(int argc, char **argv)
-{ double x; x = foo(2.0); return(0); }
-EOF
-$CC $CFLAGS -o conftest conftest.c 1>&5 2>&1
-if test -s conftest && (./conftest; exit) 2>/dev/null; then
-    lam_ac_ansi_protos=yes
-else
-    lam_ac_ansi_protos=no
-fi
-echo "$as_me:$LINENO: result: $lam_ac_ansi_protos" >&5
-echo "${ECHO_T}$lam_ac_ansi_protos" >&6
-/bin/rm -f conftest*
-if test "$lam_ac_ansi_protos" = no; then
-    { echo "$as_me:$LINENO: WARNING: *** LAM requires a compiler which understands ANSI C prototypes" >&5
-echo "$as_me: WARNING: *** LAM requires a compiler which understands ANSI C prototypes" >&2;}
-    { { echo "$as_me:$LINENO: error: *** Cannot continue." >&5
-echo "$as_me: error: *** Cannot continue." >&2;}
-   { (exit 1); exit 1; }; }
-fi
 
 # Do we have a broken compiler that does not defined __STDC__ even
 # though it really is?  Hard coded for the moment for Solaris cc.  can
@@ -5197,7 +5171,7 @@ lam_ac_size=-1
 fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
-
+lam_ac_size=$GLUSTER_SIZEOF_CHAR
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5271,6 +5245,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_SHORT
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5344,6 +5320,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_WCHAR
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5417,6 +5395,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_INT
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5490,6 +5470,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_LONG
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5563,6 +5545,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_LONG_LONG
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5636,6 +5620,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_FLOAT
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5709,6 +5695,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_DOUBLE
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5782,6 +5770,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_LONG_DOUBLE
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5855,6 +5845,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_VOID_P
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -5940,6 +5932,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_CHAR
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -6017,6 +6011,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_SHORT
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -6094,6 +6090,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_WCHAR
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -6171,6 +6169,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_INT
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -6248,6 +6248,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_LONG
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -6325,6 +6327,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_LONG_LONG
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -6402,6 +6406,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_FLOAT
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -6479,6 +6485,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_DOUBLE
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -6556,6 +6564,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_LONG_DOUBLE
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -6633,6 +6643,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_VOID_P
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -8862,6 +8874,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_CXX_BOOL
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -8950,6 +8964,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=$GLUSTER_SIZEOF_CXX_INT
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -9048,6 +9064,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_CXX_BOOL
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -9140,6 +9158,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_align=$GLUSTER_ALIGNMENT_CXX_INT
+
 if test "`expr $lam_ac_align \<= 0`" = "1"; then
     { echo "$as_me:$LINENO: WARNING: *** Problem running configure test!" >&5
 echo "$as_me: WARNING: *** Problem running configure test!" >&2;}
@@ -9379,8 +9399,8 @@ lam_fortran_underscore=0
 lam_fortran_caps=0
 lam_fortran_no_underscore=0
 
-LAM_SIZEOF_FORTRAN_INT=0
-LAM_SIZEOF_FORTRAN_REAL=0
+LAM_SIZEOF_FORTRAN_INT=4
+LAM_SIZEOF_FORTRAN_REAL=4
 LAM_SIZEOF_FORTRAN_DBLPREC=0
 LAM_SIZEOF_FORTRAN_COMPLEX=0
 LAM_SIZEOF_FORTRAN_DBLCOMPLEX=0
@@ -9719,6 +9739,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_SIZEOF_FORTRAN_INT > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_fortsize=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_fortsize" >&5
@@ -9923,6 +9946,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_SIZEOF_FORTRAN_REAL > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_fortsize=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_fortsize" >&5
@@ -10127,6 +10153,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_SIZEOF_FORTRAN_DBLPREC > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_fortsize=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_fortsize" >&5
@@ -10331,6 +10360,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_SIZEOF_FORTRAN_COMPLEX > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_fortsize=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_fortsize" >&5
@@ -10535,6 +10567,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_SIZEOF_FORTRAN_DBLCOMPLEX > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_fortsize=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_fortsize" >&5
@@ -10751,6 +10786,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_ALIGNMENT_FORTRAN_INT > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_align=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_align" >&5
@@ -10961,6 +10999,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_ALIGNMENT_FORTRAN_REAL > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_align=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_align" >&5
@@ -11171,6 +11212,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_ALIGNMENT_FORTRAN_DBLPREC > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_align=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_align" >&5
@@ -11381,6 +11425,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_ALIGNMENT_FORTRAN_COMPLEX > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_align=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_align" >&5
@@ -11591,6 +11638,9 @@ else
     HAPPY=0
 fi
 
+HAPPY=1
+echo $GLUSTER_ALIGNMENT_FORTRAN_DBLCOMPLEX > conftestval
+
 if test "$HAPPY" = "1" -a -f conftestval; then
     lam_ac_align=`cat conftestval`
     echo "$as_me:$LINENO: result: $lam_ac_align" >&5
@@ -18928,73 +18978,7 @@ _ACEOF
 # and suggesting fixes.
 
 
-# glibc's memcpy has weird copying issues when the size being copied
-# is not divisble by 4 on x86 platforms.  See if we're linking against
-# glibc, and if so, enable a workaround that first copies (size % 8)
-# "prefix" bytes over before memcpy'ing over the rest of the block.
-# Also provide a --without option to turn this behavior off because we
-# don't know when/if glibc will fix this behavior, and/or if it really
-# happens on all platforms.
-
-echo "$as_me:$LINENO: checking for glibc" >&5
-echo $ECHO_N "checking for glibc... $ECHO_C" >&6
-if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
-else
-  cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-
-#include <stdio.h>
-
-int main(int argc, char* argv[])
-{
-  FILE *fp = fopen("configtest.out", "w");
-#ifdef __GLIBC__
-  fprintf(fp, "1\n");
-#else
-  fprintf(fp, "0\n");
-#endif
-  fclose(fp);
-  return 0;
-}
-_ACEOF
-rm -f conftest$ac_exeext
-if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
-  (eval $ac_link) 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } && { ac_try='./conftest$ac_exeext'
-  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
-  (eval $ac_try) 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; }; then
-  :
-else
-  echo "$as_me: program exited with status $ac_status" >&5
-echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-fi
-rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
-fi
-have_glibc=`cat configtest.out`
-if test "$have_glibc" = "1"; then
-    echo "$as_me:$LINENO: result: yes" >&5
-echo "${ECHO_T}yes" >&6
-else
-    echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6
-fi
-rm -f configtest.out
+have_glibc=1
 
 # If we have glibc, then assume we want the prefix memcpy
 
@@ -19810,6 +19794,9 @@ _ACEOF
 #
 # The final "test", just to make it obvious what happened
 #
+
+lam_have_sysv_sem=1
+
 echo "$as_me:$LINENO: checking for SYSV semaphore support" >&5
 echo $ECHO_N "checking for SYSV semaphore support... $ECHO_C" >&6
 if test "$lam_have_sysv_sem" = "1"; then
@@ -20129,6 +20116,9 @@ fi
 #
 # The final "test", just to make it obvious what happened
 #
+
+lam_have_sysv_shmem=1
+
 echo "$as_me:$LINENO: checking for SYSV shared memory support" >&5
 echo $ECHO_N "checking for SYSV shared memory support... $ECHO_C" >&6
 if test "$lam_have_sysv_shmem" = "1"; then
@@ -22366,77 +22356,9 @@ _ACEOF
 
 echo "$as_me:$LINENO: checking if threads have different pids (pthreads on linux)" >&5
 echo $ECHO_N "checking if threads have different pids (pthreads on linux)... $ECHO_C" >&6
-CPPFLAGS_save="$CPPFLAGS"
-CPPFLAGS="$CPPFLAGS $THREAD_CPPFLAGS"
-CFLAGS_save="$CFLAGS"
-CFLAGS="$CFLAGS $THREAD_CFLAGS"
-LDFLAGS_save="$LDFLAGS"
-LDFLAGS="$LDFLAGS $THREAD_LDFLAGS"
-LIBS_save="$LIBS"
-LIBS="$LIBS $THREAD_LIBS"
-if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
-else
-  cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-#include <pthread.h>
-#include <sys/types.h>
-#include <unistd.h>
-#include <stdlib.h>
-void *checkpid(void *arg);
-int main(int argc, char* argv[]) {
-  pthread_t thr;
-  int pid, retval;
-  pid = getpid();
-  pthread_create(&thr, NULL, checkpid, &pid);
-  pthread_join(thr, (void **) &retval);
-  exit(retval);
-}
-void *checkpid(void *arg) {
-   int ret;
-   int ppid = *((int *) arg);
-   if (ppid == getpid())
-     ret = 0;
-   else
-     ret = 1;
-     pthread_exit((void *) ret);
-}
-_ACEOF
-rm -f conftest$ac_exeext
-if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
-  (eval $ac_link) 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } && { ac_try='./conftest$ac_exeext'
-  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
-  (eval $ac_try) 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; }; then
-  MSG=no LAM_THREADS_HAVE_DIFFERENT_PIDS=0
-else
-  echo "$as_me: program exited with status $ac_status" >&5
-echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
 
-( exit $ac_status )
 MSG=yes LAM_THREADS_HAVE_DIFFERENT_PIDS=1
-fi
-rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
-fi
 
-CFLAGS="$CFLAGS_save"
-CPPFLAGS="$CPPFLAGS_save"
-LDFLAGS="$LDFLAGS_save"
-LIBS="$LIBS_save"
 
 echo "$as_me:$LINENO: result: $MSG" >&5
 echo "${ECHO_T}$MSG" >&6
@@ -25467,6 +25389,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=4
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
@@ -25540,6 +25464,8 @@ fi
 rm -f core *.core gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
 fi
 
+lam_ac_size=1
+
 echo "$as_me:$LINENO: result: $lam_ac_size" >&5
 echo "${ECHO_T}$lam_ac_size" >&6
 if test "`expr $lam_ac_size \<= 0`" = "1"; then
