diff -pruN termcap-1.3.1/Makefile.in termcap-patch/Makefile.in
--- termcap-1.3.1/Makefile.in	1995-08-16 17:54:29.000000000 -0700
+++ termcap-patch/Makefile.in	2005-07-17 09:07:21.366639448 -0700
@@ -21,7 +21,7 @@ srcdir = @srcdir@
 VPATH = @srcdir@
 
 CC = @CC@
-AR = ar
+AR ?= ar
 RANLIB = @RANLIB@
 
 INSTALL = @INSTALL@
@@ -72,30 +72,31 @@ all:	libtermcap.a info
 	$(CC) -c $(CPPFLAGS) $(DEFS) -I. -I$(srcdir) $(CFLAGS) $<
 
 install: all installdirs @installdata@
-	$(INSTALL_DATA) libtermcap.a $(libdir)/libtermcap.a
-	-$(RANLIB) $(libdir)/libtermcap.a
-	cd $(srcdir); $(INSTALL_DATA) termcap.h $(includedir)/termcap.h
+	$(INSTALL_DATA) libtermcap.a $(DESTDIR)/$(libdir)/libtermcap.a
+	-$(RANLIB) $(DESTDIR)/$(libdir)/libtermcap.a
+	cd $(srcdir); $(INSTALL_DATA) termcap.h $(DESTDIR)/$(includedir)/termcap.h
 	-cd $(srcdir); test -z "$(oldincludedir)" || \
 	  $(INSTALL_DATA) termcap.h $(oldincludedir)/termcap.h
 	cd $(srcdir); for f in termcap.info*; \
-	do $(INSTALL_DATA) $$f $(infodir)/$$f; done
+	do $(INSTALL_DATA) $$f $(DESTDIR)/$(infodir)/$$f; done
 
 uninstall: @uninstalldata@
-	rm -f $(libdir)/libtermcap.a $(includedir)/termcap.h
+	rm -f $(DESTDIR)/$(libdir)/libtermcap.a $(DESTDIR)/$(includedir)/termcap.h
 	test -z "$(oldincludedir)" || rm -f $(oldincludedir)/termcap.h
-	rm -f $(infodir)/termcap.info*
+	rm -f $(DESTDIR)/$(infodir)/termcap.info*
 
 # These are separate targets to avoid trashing the user's existing
 # termcap file unexpectedly.
 install-data:
-	$(INSTALL_DATA) ${srcdir}/termcap.src ${termcapfile}
+	$(INSTALL_DATA) ${srcdir}/termcap.src $(DESTDIR)/${termcapfile}
 
 uninstall-data:
 	rm -f ${termcapfile}
 
 installdirs:
-	$(SHELL) ${srcdir}/mkinstalldirs $(bindir) $(libdir) \
-	$(includedir) $(infodir)
+	$(SHELL) ${srcdir}/mkinstalldirs $(DESTDIR)/$(bindir) \
+	$(DESTDIR)/$(libdir) \
+	$(DESTDIR)/$(includedir) $(DESTDIR)/$(infodir)
 
 Makefile: Makefile.in config.status
 	$(SHELL) config.status
