diff -pruN atftp-0.7/argz.h atftp-0.7-h/argz.h
--- atftp-0.7/argz.h	2003-01-21 07:08:35.000000000 +0530
+++ atftp-0.7-h/argz.h	2006-02-02 18:16:34.784852000 +0530
@@ -180,7 +180,7 @@ extern char *argz_next (__const char *__
 #ifdef __USE_EXTERN_INLINES
 extern inline char *
 __argz_next (__const char *__argz, size_t __argz_len,
-	     __const char *__entry) __THROW
+	     __const char *__entry)
 {
   if (__entry)
     {
@@ -194,7 +194,7 @@ __argz_next (__const char *__argz, size_
 }
 extern inline char *
 argz_next (__const char *__argz, size_t __argz_len,
-	   __const char *__entry) __THROW
+	   __const char *__entry)
 {
   return __argz_next (__argz, __argz_len, __entry);
 }
